python/paddle/fluid/dygraph/dygraph_to_static/loop_transformer.py

```diff
@@ -309,11 +316,59 @@ def _node_var_type_is_basic(self, node_var_type):
         return False
 
     def _is_call_func_name_node(self, node):
-        parent_node = self.node_to_wrapper_map[node].parent.node
+        parent_node = self._get_parent_node(node)
         if isinstance(parent_node, gast.Call) and parent_node.func == node:
             return True
         return False
 
+    def _get_parent_node(self, node):
+        wrapper_node = self.node_to_wrapper_map.get(node)
+        if wrapper_node:
+            parent_node = wrapper_node.parent.node
+            return parent_node
+        return None
+
+    def _remove_target_vars_of_for(self,
+                                   before_or_after_loop_vars,
+                                   loop_node=None):
+        """
+        Remove target vars of gast.For from before_loop_vars or after_loop_vars.
+        :param before_or_after_loop_vars: before_loop_vars or after_loop_vars of loop_node.
+        :param loop_node: Current loop node.
+        """
+
+        removed_vars = set()
+        for name_node in before_or_after_loop_vars:
+            if not isinstance(name_node, gast.Name):
+                continue
+
+            parent_node = self._get_parent_node(name_node)
+            if isinstance(parent_node, gast.Tuple):
+                parent_node = self._get_parent_node(parent_node)
```

Comment:
Here, why we need get parent of parent node when type of parent_node is `tuple`?

Reply:
`gast.For.target` can be gast.Tuple. For example:
```Python
for i, j in enumerate(x)
```
 There are two target vars :`i` and `j` .

Reply:
You can add a `Note` comment here to make it easy to understand.

Reply:
done